compute_tools/src/compute.rs

```diff
@@ -442,8 +442,42 @@ impl ComputeNode {
 
         let pg = self.start_postgres(spec.storage_auth_token.clone())?;
 
+        // Maybe apply the spec
         if spec.spec.mode == ComputeMode::Primary {
-            self.apply_config(&compute_state)?;
+            let spec = &compute_state.pspec.as_ref().expect("spec must be set").spec;
+
+            // Get spec_id or make it up by hashing
+            //
+            // TODO Make spec_id required so there would be no need to hash.
+            let spec_id = spec.operation_uuid.clone().unwrap_or_else(|| {
+                use std::collections::hash_map::DefaultHasher;
+                use std::hash::{Hash, Hasher};
+
+                // HACK Exclude postgresql.conf because it doesn't need
+                //      to be applied like the other fields in the spec
+                let mut spec_no_conf = spec.clone();
+                spec_no_conf.cluster.postgresql_conf = None;
```

Comment:
I think we should also trim `operation_uuid`, because it's different for every start. So every new wake up of idle compute will still trigger apply

Reply:
Similar with `storage_auth_token`, I'm afraid that it's just different every start

Reply:
And I also see `cluster.settings`, which I'm not sure are set, but it's safer to remove them too before hashing

Reply:
We should maybe refactor the spec to split it into "persistent" and "local" changes.

Currently it's split into "delta operations" and "all the rest", but I see no distinction between the two. The entire spec should be idempotent.

Reply:
It is idempotent, as it's serialized into operation in db and then used as is. The only slight non-idempotency is that we mix-in auth JWT every time, which is reasonable as certificates may change